app/server/app/routes/api.js

```diff
@@ -79,10 +79,32 @@ async function queryColumnValues(profile, column, params, schema) {
     else parsedParams.filters[name] = value;
   });
 
+  // get columns for where clause
+  const columnsForFilter = [];
+  profile.columns.forEach((col) => {
+    if (parsedParams.filters.hasOwnProperty(col.alias)) {
+      columnsForFilter.push(col.name);
+    }
+  });
+
+  // search through tableconfig.materializedViews to see if the column
+  // we need is in here
+  const materializedView = profile.materializedViews.find((mv) => {
+    let hasAllColumns = mv.columns.find((mvCol) => mvCol.name === column.name);
+    columnsForFilter.forEach((col) => {
+      const column = mv.columns.find((mvCol) => mvCol.name === col);
+      if (!column) hasAllColumns = false;
+    });
+
+    if (hasAllColumns) return mv;
```

Comment:
This could be eagerly returned to save a few loops:
```suggestion
  for (const col of columnsForFilter.concat(column.name)) {
    if (!mv.columns.find((mvCol) => mvCol.name === col)) return;
  }
  return mv;
```

Reply:
Good call!